src/operator/controllers/intents_reconcilers/network_policy.go

```diff
@@ -154,6 +165,28 @@ func (r *NetworkPolicyReconciler) handleNetworkPolicyCreation(
 	return r.UpdateExistingPolicy(ctx, existingPolicy, newPolicy, intent, intentsObjNamespace)
 }
 
+func (r *NetworkPolicyReconciler) shouldProtectServer(ctx context.Context, intent otterizev1alpha2.Intent, intentObjNamespace string) (bool, error) {
```

Comment:
Can you change the signature to get otterizeServiceName instead of intent and ns?